src/test/java/teammates/test/cases/ui/pagedata/InstructorFeedbackEditPageDataTest.java

```diff
@@ -0,0 +1,319 @@
+        
+        assertEquals("You have not created any questions for this feedback session yet. Click the button below to add a feedback question.", data.getEmptyFsMsg());
+        
+        questionForms = data.getQnForms();
+        assertEquals(0, questionForms.size());
+        
+        previewForm = data.getPreviewForm();
+        assertEquals(0, previewForm.getStudentToPreviewAsOptions().size());
+        assertEquals(0, previewForm.getInstructorToPreviewAsOptions().size());
+        
+        newQuestionForm = data.getNewQnForm();
+        assertEquals(new Url(Const.ActionURIs.INSTRUCTOR_FEEDBACKS_PAGE)
+                        .withUserId(instructor.googleId)
+                        .withCourseId(fs.courseId)
+                        .withSessionName(fs.feedbackSessionName).toString(), newQuestionForm.getDoneEditingLink().toString());
+        assertEquals(false, newQuestionForm.getGeneralSettings().isNumberOfEntitiesToGiveFeedbackToChecked());
+        
+        assertEquals(new Url(Const.ActionURIs.INSTRUCTOR_FEEDBACKS_PAGE)
+                         .withUserId(instructor.googleId)
+                         .withCourseId(fs.courseId)
+                         .withSessionName(fs.feedbackSessionName).toString(),
+                     newQuestionForm.getDoneEditingLink().toString());
+        
+        
+        copyForm = data.getCopyQnForm();
+        assertEquals(0, copyForm.getQuestionRows().size());
+        
+        
+        ______TS("instructor with insufficient permissions");
+        // setup
+        data = new InstructorFeedbackEditPageData(dataBundle.accounts.get("helperOfCourse1"));
+        fs = dataBundle.feedbackSessions.get("session1InCourse1");
+        fs.resultsVisibleFromTime = Const.TIME_REPRESENTS_FOLLOW_VISIBLE;
+        fs.sessionVisibleFromTime = Const.TIME_REPRESENTS_FOLLOW_OPENING;
+        
+        questions = new ArrayList<FeedbackQuestionAttributes>();
+        questions.add(dataBundle.feedbackQuestions.get("qn1InSession1InCourse1"));
+        questions.add(dataBundle.feedbackQuestions.get("qn2InSession1InCourse1"));
+        
+        copiableQuestions = new ArrayList<FeedbackQuestionAttributes>();
+        copiableQuestions.addAll(dataBundle.feedbackQuestions.values());
+        
+        questionHasResponses = new HashMap<String, Boolean>();
+        questionHasResponses.put(dataBundle.feedbackQuestions.get("qn1InSession1InCourse1").getId(), true);
+        
+        studentList = new ArrayList<StudentAttributes>();
+        studentList.add(dataBundle.students.get("student1InCourse1"));
+        
+        instructorList = new ArrayList<InstructorAttributes>();
+        instructorList.add(dataBundle.instructors.get("instructor1OfCourse1"));
+        
+        instructor = dataBundle.instructors.get("helperOfCourse1");
+        instructor.privileges = gson.fromJson(instructor.instructorPrivilegesAsText, InstructorPrivileges.class);
```

Comment:
Abstract out a method for setting the instructor privileges for getting instructor attributes out of the data bundle